meshtastic/interfaces/ble/interface.py

```
@@ -627,6 +630,104 @@ def _handle_malformed_fromnum(self, reason: str, exc_info: bool = False) -> None
+                report_exception = cast(Callable[[str], Any], hook)
+                break
+        if report_exception is not None:
+            try:
+                report_exception(error_msg)
+            except Exception:  # noqa: BLE001 - callback error reporting is best effort
+                logger.debug(error_msg, exc_info=True)
+            return
+        logger.debug(error_msg, exc_info=True)
+
+    @staticmethod
+    def _invoke_safe_execute_compat(
```

Ruff (0.15.6)

[warning] Too many return statements (7 > 6)

(PLR0911)